tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -24,23 +24,278 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
+import java.nio.*;
+
 /** Unit tests for {@link org.tensorflow.Tensor}. */
 @RunWith(JUnit4.class)
 public class TensorTest {
+  private static final double EPSILON = 1e-7;
+  private static final float EPSILON_F = 1e-7f;
+
+  @Test
+  @SuppressWarnings("unused")
+  public void createWithBuffer() {
+    int[] ints = {1, 2, 3, 4};
+    float[] floats = {1f, 2f, 3f, 4f};
+    double[] doubles = {1d, 2d, 3d, 4d};
+    long[] longs = {1L, 2L, 3L, 4L};
+    boolean[] bools = {true, false, true, false};
+    long[] shape = {4};
+
+    // validate creating a tensor using a direct byte buffer (in host order)
```

Comment:
Would it make sense to split these into separate tests?

1. "byte" tests - like use of the direct buffer and the endian-ness flips

2. One for all the typed buffers (lines 68-91)

3. One for shape checking (lines 92-128)

4. One for datatype checking